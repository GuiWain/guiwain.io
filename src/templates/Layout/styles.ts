import styled, { css } from 'styled-components'

import UseMenu from 'components/Menu'
import UseFooter from 'components/Footer'

export const Menu = styled(UseMenu)``
export const Footer = styled(UseFooter)``

export const Layout = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.primaryBg};
    width: 100%;
    height: 100vh;

    padding-top: 2rem;

    transition: background ${theme.transition.default};
  `}
`
