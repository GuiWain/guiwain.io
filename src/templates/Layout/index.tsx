import * as S from './styles'
import * as t from 'styles/theme'

import { parseCookies, setCookie } from 'nookies'
import { ThemeProvider } from 'styled-components'
import { Theme } from 'components/Menu'
import { useEffect, useState } from 'react'

export type LayoutProps = {
  children: JSX.Element
}

const Layout = ({ children }: LayoutProps) => {
  const [theme, setTheme] = useState(t.light)

  const handleTheme = (theme: Theme) => {
    const newTheme = theme === 'light' ? t.light : t.dark
    setTheme(newTheme)
  }

  useEffect(() => {
    const cookies = parseCookies()

    if (!cookies.theme) {
      const initialTheme = 'light'
      const ONE_MONTH = 60 * 60 * 24 * 4

      setCookie(null, 'theme', initialTheme, {
        maxAge: ONE_MONTH
      })

      return
    }

    const currentyTheme = cookies.theme
    const updatedTheme = currentyTheme === 'light' ? t.light : t.dark
    setTheme(updatedTheme)
  }, [])

  return (
    <ThemeProvider theme={theme}>
      <S.Layout>
        <S.Menu
          getTheme={(theme) => handleTheme(theme)}
          checked={theme.type === 'dark'}
        />

        {children}

        <S.Footer />
      </S.Layout>
    </ThemeProvider>
  )
}

export default Layout
