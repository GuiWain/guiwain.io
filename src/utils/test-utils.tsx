import { ThemeProvider } from 'styled-components'
import { render, RenderResult } from '@testing-library/react'
import * as theme from 'styles/theme'

const renderWithTheme = (children: React.ReactNode): RenderResult =>
  render(<ThemeProvider theme={theme.light}>{children}</ThemeProvider>)

export * from '@testing-library/react'
export { renderWithTheme as render }
