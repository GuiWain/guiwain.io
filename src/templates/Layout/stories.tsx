import { ComponentMeta, ComponentStory } from '@storybook/react'
import Layout from '.'

export default {
  title: 'Templates/Layout',
  component: Layout
} as ComponentMeta<typeof Layout>

export const Default: ComponentStory<typeof Layout> = () => (
  <Layout>
    <h1>Children</h1>
  </Layout>
)
