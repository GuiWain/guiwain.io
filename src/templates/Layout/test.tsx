import { render } from 'utils/test-utils'

import Layout from '.'

describe('<Layout />', () => {
  it('should render the Layout', () => {
    const { container } = render(
      <Layout>
        <p>lalala</p>
      </Layout>
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
