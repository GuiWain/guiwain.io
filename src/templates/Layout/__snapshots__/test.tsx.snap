// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render the Layout 1`] = `
.c3 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c4 {
  color: #FFEA00;
  width: 1.6rem;
}

.c1 {
  margin: auto;
  width: 90%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .ant-switch {
  width: 4rem;
  height: 2.2rem;
  background-color: #2D2529;
}

.c1 .ant-switch:focus-within,
.c1 .ant-switch:focus-visible {
  outline: none;
}

.c1 .ant-switch:focus {
  box-shadow: 0rem 0rem 0.2rem 0.1rem rgba(122,0,65,0.2);
}

.c1 .ant-switch:focus .ant-switch-handle:before {
  box-shadow: inset 0rem 0rem 0.2rem 0.1rem rgba(122,0,65,0.3);
}

.c1 .ant-switch-handle:before {
  background-color: #F9F9F9;
}

.c1 .ant-switch-checked .ant-switch-inner {
  margin: 0rem 2.5rem 0rem 0.2rem;
}

.c2 {
  color: #2D2529;
  font-family: Montserrat,sans-serif;
  font-size: 3.2rem;
  font-weight: 900;
  cursor: pointer;
}

.c2:hover {
  color: #7A0041;
}

.c5 {
  background: #ADADAD;
  box-shadow: 0px 0px 1px 0px rgba(122,0,65,0.3);
  -webkit-transition: background 0.4s ease-in-out,box-shadow 0.4s ease-in-out;
  transition: background 0.4s ease-in-out,box-shadow 0.4s ease-in-out;
  width: 100%;
  height: 5rem;
  position: absolute;
  bottom: 0;
  left: 0;
}

.c5:hover {
  box-shadow: 0px -1px 3px 0px rgba(122,0,65,0.2);
}

.c6 {
  width: 90%;
  height: 100%;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  background: #F9F9F9;
  width: 100%;
  height: 100vh;
  padding-top: 2rem;
  -webkit-transition: background 0.4s ease-in-out;
  transition: background 0.4s ease-in-out;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <a
      class="c2"
    >
      GuiWain
    </a>
    <button
      aria-checked="false"
      class="ant-switch "
      role="switch"
      type="button"
    >
      <div
        class="ant-switch-handle"
      />
      <span
        class="ant-switch-inner"
      >
        <svg
          aria-hidden="true"
          class="c3 c4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="24"
            width="24"
          />
          <path
            d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37a.996.996 0 00-1.41 0 .996.996 0 000 1.41l1.06 1.06c.39.39 1.03.39 1.41 0a.996.996 0 000-1.41l-1.06-1.06zm1.06-10.96a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36a.996.996 0 000-1.41.996.996 0 00-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
          />
        </svg>
      </span>
    </button>
  </div>
  <p>
    lalala
  </p>
  <div
    class="c5"
  >
    <nav
      class="c6"
    >
      <svg
        aria-hidden="true"
        class="c3 "
        fill="currentColor"
        focusable="false"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M363 176L246 464h47.24l24.49-58h90.54l24.49 58H480zm-26.69 186L363 279.85 389.69 362zM272 320c-.25-.19-20.59-15.77-45.42-42.67 39.58-53.64 62-114.61 71.15-143.33H352V90H214V48h-44v42H32v44h219.25c-9.52 26.95-27.05 69.5-53.79 108.36-32.68-43.44-47.14-75.88-47.33-76.22L143 152l-38 22 6.87 13.86c.89 1.56 17.19 37.9 54.71 86.57.92 1.21 1.85 2.39 2.78 3.57-49.72 56.86-89.15 79.09-89.66 79.47L64 368l23 36 19.3-11.47c2.2-1.67 41.33-24 92-80.78 24.52 26.28 43.22 40.83 44.3 41.67L255 362z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="c3 "
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19.536 9.146c-1.373 0-2.133 1.014-2.294 2.116h4.608c-.125-1.05-.867-2.115-2.314-2.115m-2.26 3.617c.235 1.156 1.193 1.97 2.532 1.97.725 0 1.77-.27 2.384-.914l1.175 1.35c-1.064 1.11-2.653 1.426-3.74 1.426-2.64 0-4.697-1.906-4.697-4.606 0-2.535 1.894-4.62 4.57-4.62 2.585 0 4.5 1.98 4.5 4.604v.766h-6.723v.023zm-6.487 3.83v-5.69c0-.976-.435-1.536-1.338-1.536-.814 0-1.355.585-1.717 1.007v6.24h-2.35v-5.7c0-.976-.415-1.532-1.318-1.532-.813 0-1.375.586-1.717 1.006v6.24H0V7.505h2.35v1.15c.4-.463 1.302-1.26 2.71-1.26 1.247 0 2.096.526 2.477 1.59.524-.761 1.5-1.59 2.91-1.59 1.7 0 2.69 1.01 2.69 2.963v6.24h-2.353l.005-.007z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="c3 "
        fill="none"
        focusable="false"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="22 3 2 3 10 12.46 10 19 14 21 14 12.46 22 3"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="c3 "
        fill="none"
        focusable="false"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="10"
        />
        <polyline
          points="16 12 12 8 8 12"
        />
        <line
          x1="12"
          x2="12"
          y1="16"
          y2="8"
        />
      </svg>
    </nav>
  </div>
</div>
`;
